a lot of warning in inference related codes

- anakin related codes: https://paddleci.ngrok.io/viewLog.html?buildId=1413&buildTypeId=Paddle_PrCi&tab=buildLog&_focus=3106#_state=67
```
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/parameter.h:26:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/data_types.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/any.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/node.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:169:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (shape.dims() != TensorAPI::layout_dims::value || \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:201:13: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]              //if (!(_valid_shape + _offset <= _shape)) { \
[13:19:08][Step 1/1]              ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:252:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (shape.dims() != TensorAPI::layout_dims::value || \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:283:13: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]              //if (!(_valid_shape + _offset <= _shape)) { \
[13:19:08][Step 1/1]              ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:313:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (start < 0) { \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:316:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (end > dims()) { \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:336:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (start < 0) { \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:339:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (end > dims()) { \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/parameter.h:26:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/data_types.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/any.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/node.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:552:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (valid_shape.dims() != TensorAPI::layout_dims::value \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:578:9: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]          //if (valid_size() != tensor.valid_size()) { \
[13:19:08][Step 1/1]          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:700:53: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]              int idx_dst = (i % ratio_dst) * cpy_len;//off_dst[abs(axis_discontinue_dst)] * \
[13:19:08][Step 1/1]                                                      ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:708:53: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]              int idx_src = (i % ratio_src) * cpy_len;//off_src[abs(axis_discontinue_src)] * \
[13:19:08][Step 1/1]                                                      ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:852:53: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]              int idx_dst = (i % ratio_dst) * cpy_len;//off_dst[abs(axis_discontinue_dst)] * \
[13:19:08][Step 1/1]                                                      ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:860:53: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]              int idx_src = (i % ratio_src) * cpy_len;//off_src[abs(axis_discontinue_src)] * \
[13:19:08][Step 1/1]                                                      ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/net/operator_func.h:20:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:20,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:215:1: warning: multi-line comment [-Wcomment]
[13:19:08][Step 1/1]  //#define ANAKIN_REGISTER_OP(OpName) \
[13:19:08][Step 1/1]  ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/utils/logger/logger.h:22:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/arc.h:23,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph_base.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h: In function 'std::__cxx11::string logger::core::funcRegister::stacktrace_as_stdstring(int)':
[13:19:08][Step 1/1] Install the project...
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h:743:29: warning: too many arguments for format [-Wformat-extra-args]
[13:19:08][Step 1/1]                       sourcep);
[13:19:08][Step 1/1]                              ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h:743:29: warning: too many arguments for format [-Wformat-extra-args]
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/utils/logger/logger.h:22:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/arc.h:23,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph_base.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h:750:21: warning: unused variable 'endIndex' [-Wunused-variable]
[13:19:08][Step 1/1]                  int endIndex = fread(pbuffer, sizeof(pbuffer), 1, pf);;
[13:19:08][Step 1/1]                      ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/utils/logger/logger.h:22:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/arc.h:23,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph_base.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h: In function 'void logger::core::funcRegister::get_thread_name(char*, long long unsigned int, bool)':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h:847:89: warning: field width specifier '*' expects argument of type 'int', but argument 4 has type 'long long unsigned int' [-Wformat=]
[13:19:08][Step 1/1]              snprintf(buffer, length, "%*X", length - 1, static_cast<unsigned>(thread_id));
[13:19:08][Step 1/1]                                                                                          ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/utils/logger/logger_core.h:847:89: warning: field width specifier '*' expects argument of type 'int', but argument 4 has type 'long long unsigned int' [-Wformat=]
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/saber/core/shape.h:19:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/saber/core/tensor.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/parameter.h:26,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/data_types.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/any.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/node.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/common.h: In function 'const char* anakin::saber::saber_get_error_string(anakin::saber::SaberStatus)':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/common.h:40:12: warning: enumeration value 'SaberWrongDevice' not handled in switch [-Wswitch]
[13:19:08][Step 1/1]      switch (error_code) {
[13:19:08][Step 1/1]             ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/net/operator_func.h:20:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:20,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h: In member function 'virtual anakin::Status anakin::OperatorHelper<Ttype, Dtype, Ptype>::InitParam()':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:86:5: warning: no return statement in function returning non-void [-Wreturn-type]
[13:19:08][Step 1/1]      }
[13:19:08][Step 1/1]      ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h: In member function 'virtual anakin::Status anakin::OperatorHelper<Ttype, Dtype, Ptype>::Init(anakin::OpContext<Ttype>&, const std::vector<anakin::saber::Tensor<TargetType, datatype, anakin::saber::NCHW>*>&, std::vector<anakin::saber::Tensor<TargetType, datatype, anakin::saber::NCHW>*>&)':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:95:5: warning: no return statement in function returning non-void [-Wreturn-type]
[13:19:08][Step 1/1]      }
[13:19:08][Step 1/1]      ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h: In member function 'virtual anakin::Status anakin::OperatorHelper<Ttype, Dtype, Ptype>::InferShape(const std::vector<anakin::saber::Tensor<TargetType, datatype, anakin::saber::NCHW>*>&, std::vector<anakin::saber::Tensor<TargetType, datatype, anakin::saber::NCHW>*>&)':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:103:5: warning: no return statement in function returning non-void [-Wreturn-type]
[13:19:08][Step 1/1]      }
[13:19:08][Step 1/1]      ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:22:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/operator_func.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:20,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/factory.h: In instantiation of 'class anakin::ObjectRegister<anakin::OpAttrWarpper>':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:170:37:   required from here
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/factory.h:146:7: warning: 'class anakin::ObjectRegister<anakin::OpAttrWarpper>' has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
[13:19:08][Step 1/1]  class ObjectRegister : public ObjectRegisterBase<PolicyType, std::string> {
[13:19:08][Step 1/1]        ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/net/operator_func.h:20:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:20,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h: At global scope:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:170:7: warning: base class 'class anakin::ObjectRegister<anakin::OpAttrWarpper>' has accessible non-virtual destructor [-Wnon-virtual-dtor]
[13:19:08][Step 1/1]  class OpAttrObjectRegister : public ObjectRegister<OpAttrWarpper> {
[13:19:08][Step 1/1]        ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/core/operator/operator.h:170:7: warning: 'class anakin::OpAttrObjectRegister' has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/core/parameter.h:26:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/data_types.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/any.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/node.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:20,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h: In instantiation of 'int anakin::saber::Tensor<TargetType, datatype, LayeOutType>::start_index() const [with TargetType = anakin::saber::TargetType<(anakin::saber::TargetTypeEnum)1>; anakin::saber::DataType datatype = (anakin::saber::DataType)1; LayOutType = anakin::saber::NCHW]':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:499:75:   required from 'anakin::saber::Tensor<TargetType, datatype, LayeOutType>::Dtype* anakin::saber::Tensor<TargetType, datatype, LayeOutType>::mutable_data(int) [with TargetType = anakin::saber::TargetType<(anakin::saber::TargetTypeEnum)1>; anakin::saber::DataType datatype = (anakin::saber::DataType)1; LayOutType = anakin::saber::NCHW; anakin::saber::Tensor<TargetType, datatype, LayeOutType>::Dtype = float]'
[13:19:08][Step 1/1] [ 10%] Built target threadpool
[13:19:08][Step 1/1] /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:49:51:   required from here
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/saber/core/tensor.h:924:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[13:19:08][Step 1/1]          for (int i = 0; i < stride.size(); ++i) {
[13:19:08][Step 1/1]                            ^
[13:19:08][Step 1/1] In file included from /paddle/build/third_party/install/anakin/framework/graph/graph_base.h:159:0,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/graph/graph.h:19,
[13:19:08][Step 1/1]                  from /paddle/build/third_party/install/anakin/framework/core/net/net.h:19,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.h:25,
[13:19:08][Step 1/1]                  from /paddle/paddle/contrib/inference/paddle_inference_api_anakin_engine.cc:15:
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/graph/graph_base.inl: In member function 'ArcType& anakin::graph::GraphBase<VertexNameType, VertexType, WeightType, ArcType>::get_arc(VertexNameType, VertexNameType) [with VertexNameType = std::__cxx11::basic_string<char>; VertexType = std::shared_ptr<anakin::graph::Node<anakin::saber::TargetType<(anakin::saber::TargetTypeEnum)1>, (anakin::saber::DataType)1, (anakin::Precision)0> >; WeightType = anakin::saber::Tensor<anakin::saber::TargetType<(anakin::saber::TargetTypeEnum)1>, (anakin::saber::DataType)1, anakin::saber::NCHW>*; ArcType = anakin::graph::Edge<anakin::saber::TargetType<(anakin::saber::TargetTypeEnum)1>, (anakin::saber::DataType)1>]':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/graph/graph_base.inl:203:1: warning: control reaches end of non-void function [-Wreturn-type]
[13:19:08][Step 1/1]  }
[13:19:08][Step 1/1]  ^
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/graph/graph_base.inl: In member function 'ArcType& anakin::graph::GraphBase<VertexNameType, VertexType, WeightType, ArcType>::get_arc(VertexNameType, VertexNameType) [with VertexNameType = std::__cxx11::basic_string<char>; VertexType = anakin::graph::node; WeightType = anakin::graph::io; ArcType = anakin::graph::Arc<std::__cxx11::basic_string<char>, anakin::graph::io>]':
[13:19:08][Step 1/1] /paddle/build/third_party/install/anakin/framework/graph/graph_base.inl:203:1: warning: control reaches end of non-void function [-Wreturn-type]
[13:19:08][Step 1/1]  }
[13:19:08][Step 1/1]  ^
```
- test_paddle_inference_api_impl.cc: https://paddleci.ngrok.io/viewLog.html?buildId=1413&buildTypeId=Paddle_PrCi&tab=buildLog&_focus=5076#_state=67
```
[13:25:57][Step 1/1] In file included from /paddle/paddle/contrib/inference/test_paddle_inference_api_impl.cc:16:0:
[13:25:57][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h: In instantiation of 'testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = long int; T2 = long unsigned int]':
[13:25:57][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h:1421:23:   required from 'static testing::AssertionResult testing::internal::EqHelper<lhs_is_null_literal>::Compare(const char*, const char*, const T1&, const T2&) [with T1 = long int; T2 = long unsigned int; bool lhs_is_null_literal = false]'
[13:25:57][Step 1/1] /paddle/paddle/contrib/inference/test_paddle_inference_api_impl.cc:252:282:   required from here
[13:25:57][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h:1392:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[13:25:57][Step 1/1]    if (lhs == rhs) {
[13:25:57][Step 1/1]            ^
```
- subgraph_splitter_tester.cc: https://paddleci.ngrok.io/viewLog.html?buildId=1413&buildTypeId=Paddle_PrCi&tab=buildLog&_focus=4861#_state=67
```
[13:25:33][Step 1/1] In file included from /paddle/paddle/fluid/inference/analysis/ut_helper.h:17:0,
[13:25:33][Step 1/1]                  from /paddle/paddle/fluid/inference/analysis/subgraph_splitter_tester.cc:16:
[13:25:33][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h: In instantiation of 'testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int]':
[13:25:33][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h:1421:23:   required from 'static testing::AssertionResult testing::internal::EqHelper<lhs_is_null_literal>::Compare(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int; bool lhs_is_null_literal = false]'
[13:25:33][Step 1/1] /paddle/paddle/fluid/inference/analysis/subgraph_splitter_tester.cc:85:210:   required from here
[13:25:33][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h:1392:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[13:25:33][Step 1/1]    if (lhs == rhs) {
[13:25:33][Step 1/1]            ^
```
- fluid_to_data_flow_graph_pass_tester.cc: https://paddleci.ngrok.io/viewLog.html?buildId=1413&buildTypeId=Paddle_PrCi&tab=buildLog&_focus=4848#_state=67
```
[13:25:32][Step 1/1] In file included from /paddle/paddle/fluid/inference/analysis/fluid_to_data_flow_graph_pass_tester.cc:17:0:
[13:25:32][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h: In instantiation of 'testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int]':
[13:25:32][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h:1421:23:   required from 'static testing::AssertionResult testing::internal::EqHelper<lhs_is_null_literal>::Compare(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int; bool lhs_is_null_literal = false]'
[13:25:32][Step 1/1] /paddle/paddle/fluid/inference/analysis/fluid_to_data_flow_graph_pass_tester.cc:30:247:   required from here
[13:25:32][Step 1/1] /paddle/build/third_party/install/gtest/include/gtest/gtest.h:1392:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[13:25:32][Step 1/1]    if (lhs == rhs) {
```
- listen_and_serv_op.cc: https://paddleci.ngrok.io/viewLog.html?buildId=1413&buildTypeId=Paddle_PrCi&tab=buildLog&_focus=4122#_state=67
```
[13:22:11][Step 1/1] /paddle/paddle/fluid/operators/listen_and_serv_op.cc: In member function 'void paddle::operators::ListenAndServOp::RunSyncLoop(paddle::framework::Executor*, paddle::framework::ProgramDesc*, paddle::framework::Scope*, const std::vector<int>&) const':
[13:22:11][Step 1/1] /paddle/paddle/fluid/operators/listen_and_serv_op.cc:108:29: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[13:22:11][Step 1/1]    for (int blkid = 1; blkid < num_blocks; ++blkid) {
[13:22:11][Step 1/1]                              ^
```

--- a/cmake/external/anakin.cmake
+++ b/cmake/external/anakin.cmake
@@ -7,7 +7,17 @@ set(ANAKIN_INSTALL_DIR "${THIRD_PARTY_PATH}/install/anakin" CACHE PATH
 set(ANAKIN_INCLUDE "${ANAKIN_INSTALL_DIR}" CACHE STRING "root of Anakin header files")
 set(ANAKIN_LIBRARY "${ANAKIN_INSTALL_DIR}" CACHE STRING "path of Anakin library")
 
-set(ANAKIN_COMPILE_EXTRA_FLAGS -Wno-error=unused-variable -Wno-error=format-extra-args -Wno-error=comment -Wno-error=format -Wno-error=switch -Wno-error=return-type -Wno-error=non-virtual-dtor -Wno-reorder -Wno-error=cpp)
+set(ANAKIN_COMPILE_EXTRA_FLAGS 
+    -Wno-error=unused-variable -Wno-unused-variable 
+    -Wno-error=format-extra-args -Wno-format-extra-args
+    -Wno-error=comment -Wno-comment 
+    -Wno-error=format -Wno-format 
+    -Wno-error=switch -Wno-switch
+    -Wno-error=return-type -Wno-return-type 
+    -Wno-error=non-virtual-dtor -Wno-non-virtual-dtor
+    -Wno-sign-compare
+    -Wno-reorder 
+    -Wno-error=cpp)
 
 set(ANAKIN_LIBRARY_URL "https://github.com/pangge/Anakin/releases/download/3.0/anakin_release_simple.tar.gz")
 

--- a/paddle/contrib/inference/test_paddle_inference_api_impl.cc
+++ b/paddle/contrib/inference/test_paddle_inference_api_impl.cc
@@ -249,7 +249,7 @@ void MainThreadsImageClassification(bool use_gpu) {
       const size_t len = local_outputs[0].data.length();
       float* data = static_cast<float*>(local_outputs[0].data.data());
       float* ref_data = refs[tid].data<float>();
-      EXPECT_EQ(refs[tid].numel(), len / sizeof(float));
+      EXPECT_EQ((size_t)refs[tid].numel(), len / sizeof(float));
       for (int i = 0; i < refs[tid].numel(); ++i) {
         EXPECT_NEAR(ref_data[i], data[i], 1e-3);
       }

--- a/paddle/fluid/inference/analysis/fluid_to_data_flow_graph_pass_tester.cc
+++ b/paddle/fluid/inference/analysis/fluid_to_data_flow_graph_pass_tester.cc
@@ -27,7 +27,7 @@ TEST_F(DFG_Tester, Init) {
   DataFlowGraph graph;
   pass.Run(&graph);
   // Analysis is sensitive to ProgramDesc, careful to change the original model.
-  ASSERT_EQ(graph.nodes.size(), 37);
+  ASSERT_EQ(graph.nodes.size(), 37UL);
   pass.Finalize();
   LOG(INFO) << '\n' << graph.DotString();
 }

--- a/paddle/fluid/inference/analysis/subgraph_splitter_tester.cc
+++ b/paddle/fluid/inference/analysis/subgraph_splitter_tester.cc
@@ -82,7 +82,7 @@ TEST_F(DFG_Tester, Fuse) {
 
   // At least one nodes should be deleted.
   ASSERT_EQ(dfg.nodes.size(), count0 + 1);  // added a new FunctionBlock
-  ASSERT_EQ(6UL, count1);
+  ASSERT_EQ(6, count1);
 }
 
 }  // namespace analysis

